core/templates/services/user-backend-api.service.ts

```diff
@@ -0,0 +1,164 @@
+// Copyright 2018 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Service for user data.
+ */
+
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { downgradeInjectable } from '@angular/upgrade/static';
+
+import { List } from 'lodash';
+
+import { AppConstants } from 'app.constants';
+import { UserInfo, UserInfoBackendDict, UserInfoObjectFactory } from
+  'domain/user/UserInfoObjectFactory.ts';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { UrlService } from 'services/contextual/url.service.ts';
+import { WindowRef } from 'services/contextual/window-ref.service';
+
+interface SubscriptionSummary {
+  'creator_picture_data_url': string;
+  'creator_username': string;
+  'creator_impact': number;
+}
+interface PreferencesBackendDict {
+  'preferred_language_codes': string;
+  'preferred_site_language_code': string;
+  'preferred_audio_language_code': string;
+  'profile_picture_data_url': string;
+  'default_dashboard': string;
+  'user_bio': string;
+  'subject_interests': string;
+  'can_receive_email_updates': boolean;
+  'can_receive_editor_role_email': boolean;
+  'can_receive_feedback_message_email': boolean;
+  'can_receive_subscription_email': boolean;
+  'subscription_list': List<SubscriptionSummary>;
+}
+interface UrlBackendDict {
+  'login_url': string;
+}
+interface UserCommunityRightsDataBackendDict {
+  'can_review_translation_for_language_codes': boolean;
+  'can_review_voiceover_for_language_codes': boolean;
+  'can_review_questions': boolean;
+}
+
+
+@Injectable({
+  providedIn: 'root'
+})
+export class UserBackendApiService {
+  constructor(
+    private http: HttpClient,
+    private windowRef: WindowRef,
+    private urlInterpolationService: UrlInterpolationService,
+    private urlService: UrlService,
+    private userInfoObjectFactory: UserInfoObjectFactory) {}
```

Comment:
Maybe take the closing parentheses to the next line.
```suggestion
    private http: HttpClient,
    private windowRef: WindowRef,
    private urlInterpolationService: UrlInterpolationService,
    private urlService: UrlService,
    private userInfoObjectFactory: UserInfoObjectFactory
  ) {}
```

Reply:
Done!